openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -0,0 +1,190 @@
+//! This module tests the validation of proposals as defined in
+//! https://openmls.tech/book/message_validation.html#semantic-validation-of-proposals-covered-by-a-commit
+
+use openmls_rust_crypto::OpenMlsRustCrypto;
+use openmls_traits::OpenMlsCryptoProvider;
+
+use rstest::*;
+use rstest_reuse::{self, *};
+
+use crate::{
+    config::*,
+    credentials::*,
+    group::errors::*,
+    group::*,
+    key_packages::*,
+    prelude_test::{FramingParameters, MlsPlaintext, WireFormat},
```

Comment:
You should use the `framing` module here. Prelude should never be used from the inside.